src/subclassers.ts

```diff
@@ -198,3 +198,29 @@ export function WithAttached<
     };
   };
 }
+
+export function WithDoWork<TProjector extends AnyAsyncProjector>(
+  projector: TProjector
+) {
+  // tslint:disable-next-line:only-arrow-functions
+  return function<TSuper extends Constructor>(Super: TSuper) {
+    return class DoWork extends Super {
+      constructor(...args: any[]) {
+        super(...args);
+      }
```

Comment:
1. [ ] minor: change class name to WithDoWork
2. [ ] I think all this stuff below is unnecessary b/c executeHipThrustable will do exactly this anyway.  I think you can just await projector(this); Even without the try/catch

Reply:
done